[ticket/12769] Font-Awesome Integration & Icon overhaul

hanakin opened the pull request:
#### TODO:

- [x] Add CDN version of Font-Awesome
- [x] Search Icons
- [x] Navbar icons
- [x] Re-fator Drop-Downs
- [x] Re-factor Buttons
- [x] Form Icons
- [x] Post Icons

- [x] Misc Icons

- [x] Cleanup the semantics in the HTML
- [x] Cleanup /Remove unnecessary CSS
- [x] Remove replaced images
- [x] Refactor any javascript breaks
- [x] Cross browser checks
- [x] Finalize icon choices
- [x] Fix rtl
- [x] Fix responsive

Old PR: https://github.com/phpbb/phpbb/pull/2653
Ticket: https://tracker.phpbb.com/browse/PHPBB3-12769

Requirments: https://github.com/phpbb/phpbb/pull/3837

PHPBB3-12769

hanakin commented:
!set WIP

hanakin commented:
!set Styles

hanakin commented:
This is pretty much ready for review other than one small js fix for switching subscribe icons. I am going to ask for as much help with the review and cross-browser checks on this as possible as there is alot of page changes here. Not to mention alot of icons and I might have missed one

VSEphpbb commented on phpBB/styles/prosilver/template/search_results.html:
```diff
@@ -64,17 +76,36 @@ <h2 class="searchresults-title"><!-- IF SEARCH_TITLE -->{SEARCH_TITLE}<!-- ELSE
+							</a> 
+							<!-- ENDIF -->
+							<!-- IF searchresults.S_TOPIC_DELETED -->
+							<a href="{searchresults.U_MCP_QUEUE}" title="{TOPIC_DELETED">
```
missing closing bracket in TOPIC_DELETED

VSEphpbb commented:
Nice work! My main concern is the backwards compatibility issues this will introduce for extensions. Mainly in the form of where you are deleting classes used by extensions. For example, icons in the navbar.

Is it possible we can have a small compatibility layer in the CSS where some of the (deprecated) classes you eliminated are still retained so some extensions don't get too clobbered between 3.1 and 3.2? At least for `.small-icon`, and if anybody can think of any others that have been important to extensions.

VSEphpbb commented on phpBB/styles/prosilver/template/navbar_header.html:
```diff
@@ -40,50 +78,101 @@
+		<li class="dropdown-container dropdown-{S_CONTENT_FLOW_END} rightside" data-skip-responsive="true">
+			<a href="{U_VIEW_ALL_NOTIFICATIONS}" id="notification_list_button" class="dropdown-trigger">
+				<i class="icon fa-bell fa-fw"></i><span><span>{L_NOTIFICATIONS} [</span><strong>{NOTIFICATIONS_COUNT}</strong><span>]</span></span>
+			</a></span>
```
This closing span tag is a stray that has no opening

VSEphpbb commented on phpBB/styles/prosilver/template/mcp_topic.html:
```diff
@@ -112,14 +112,17 @@ <h3 id="review">
 				</p>
 				<!-- ENDIF -->
-
+	
```
Whitepsace

VSEphpbb commented on phpBB/styles/prosilver/template/posting_review.html:
```diff
@@ -14,7 +14,15 @@
 		<h3><a href="#ppr{post_review_row.POST_ID}">{post_review_row.POST_SUBJECT}</a></h3>
-		<p class="author"><!-- IF S_IS_BOT -->{post_review_row.MINI_POST_IMG}<!-- ELSE --><a href="{post_review_row.U_MINI_POST}">{post_review_row.MINI_POST_IMG}</a><!-- ENDIF --> {L_POST_BY_AUTHOR}<strong>  {post_review_row.POST_AUTHOR_FULL}</strong> &raquo; {post_review_row.POST_DATE}</p>
+		<p class="author">
+		<!-- IF S_IS_BOT -->
```
Indentation please

VSEphpbb commented on phpBB/styles/prosilver/template/posting_review.html:
```diff
@@ -14,7 +14,15 @@
+			<i class="icon fa-file fa-fw icon-lightgray icon-sm"></i><span class="sr-only">{post_review_row.MINI_POST}</span>
+		</a>
+		<!-- ENDIF -->
+		{L_POST_BY_AUTHOR}<strong>  {post_review_row.POST_AUTHOR_FULL}</strong> &raquo; {post_review_row.POST_DATE}</p>
```
Closing p tag should be on its own line

VSEphpbb commented on phpBB/styles/prosilver/template/posting_topic_review.html:
```diff
@@ -28,22 +28,30 @@ <h3 id="review">
 
-			<p class="author"><!-- IF S_IS_BOT -->{topic_review_row.MINI_POST_IMG}<!-- ELSE --><a href="{topic_review_row.U_MINI_POST}">{topic_review_row.MINI_POST_IMG}</a><!-- ENDIF --> {L_POST_BY_AUTHOR} <strong>{topic_review_row.POST_AUTHOR_FULL}</strong> &raquo; {topic_review_row.POST_DATE}  </p>
+			<p class="author">
+			<!-- IF S_IS_BOT -->
```
Indentation please and closing p tag again, same as above.

VSEphpbb commented on phpBB/styles/prosilver/template/ucp_pm_viewmessage.html:
```diff
@@ -8,148 +8,177 @@
-		<!-- IF U_VIEW_PREVIOUS_HISTORY --><a href="{U_VIEW_PREVIOUS_HISTORY}" class="left-box arrow-{S_CONTENT_FLOW_BEGIN}">{L_VIEW_PREVIOUS_HISTORY}</a><!-- ENDIF -->
-		<!-- IF U_VIEW_NEXT_HISTORY --><a href="{U_VIEW_NEXT_HISTORY}" class="right-box arrow-{S_CONTENT_FLOW_END}">{L_VIEW_NEXT_HISTORY}</a><!-- ENDIF -->
+		<!-- IF U_VIEW_PREVIOUS_HISTORY -->
+		<a href="{U_VIEW_PREVIOUS_HISTORY}" class="left-box arrow-{S_CONTENT_FLOW_BEGIN}">
```
Please indent content inside `<!-- IF -->` blocks here and elsewhere.

VSEphpbb commented on phpBB/styles/prosilver/template/viewtopic_body.html:
```diff
@@ -354,25 +384,30 @@ <h3 <!-- IF postrow.S_FIRST_ROW -->class="first"<!-- ENDIF -->><!-- IF postrow.P
-<div class="action-bar bottom">
-	<div class="buttons">
+<div class="action-bar bar-bottom">
 		<!-- EVENT viewtopic_buttons_bottom_before -->
```
Can you fix indentation levels from here down please

hanakin commented:
@VSEphpbb there is going to be a lot of backwards compatibility issues I am sure I hade to completely rewrite several things ie buttons, action-bars, serach, drop-downs to mention a few to get this to work properly so not sure small-icon is really as big an issue by comparisons.

VSEphpbb commented:
@hanakin I know .small-icon will be an issue. As for phpBB's core, we have always been trying to include as much b.c. as possible so the transition from 3.1. to 3.2 will be as smooth as possible for users, mostly with respect to extensions. Obviously there are some major changes here that may break a couple extensions in a way we can't help or forsee, but for a simple class as .small-icon and as many extensions out there that may be using (including two flagship official extension, Pages and Board Rules) I see no reason why we can't keep the old class' CSS around during the 3.2.x line.

hanakin commented:
@VSEphpbb we should probably update the official extension icons to be svg to avoid pixel icons mixing but I will look into the BC stuff, but this brings up an old discussion I remember having with a few of the others about including a deprecated.css file to handle such issues like this

VSEphpbb commented:
@hanakin We will of course update the extensions. The issue is that it currently takes several months for an extension to get validated. The reality is it will be a long time after 3.2 is released before validated extensions can catch up. We want to mitigate issues where users who choose to upgrade and find extensions are broken, and there's no update to fix it anytime soon. We really want the 3.1 to 3.2 transition to be painless. 

So yeah, a deprecated css file would be great. As I said, so far I think it only needs .small-icon, though others might think of a few other rules that would help too.

hanakin commented:
As for now I just added the deprecated class to the end of the original files with a comment as this way you do not have to worry as much about specification with in its own file.

hanakin commented:
Short of any issues anyone has with choices for the icons this is good to go as far as Browser compatability. 

It has one error that has to do with the method of inclusion of Font-Awesome being from a CDN. As for some reason there is a test to ensure all styles are coming from the theme folder which obviously fails? Need to determine how we want to handle this.

@VSEphpbb if there are any more BC stuff its probably easier to fix those in alpha/beta when users can put it through the ringers. We can add BC back in at that point.

@cyberalien when you get a chance this is a large one!

hanakin commented:
!unset WIP

VSEphpbb commented:
@hanakin I like all the icons I've seen so far except for New Post/PM Buttons. One of the fa plus symbols makes more sense than a star to me. I know we use star currently, but we don't need to keep with it IMO. Not sure how others feel.

VSEphpbb commented:
I just noticed the close button icon in Ajax dialogs needs help with positioning now. It's flush up against the sides in the corner, could use some padding/margin

cyberalien commented:
In navigation icon color doesn't match links. Its too saturated. Maybe something like this?

    .navbar a .icon {
        color: inherit;
        opacity: 0.9;
    }

Code above breaks .icon.icon-gray color, so that rule will have to be duplicated with .navbar as well.

Orange color for last post link in topics list doesn't match layout. All icons are red, orange link near red icon looks rather bad. It is also way too small, but making it larger looks equally bad. Maybe use fa-file-o instead of fa-file for it?

Similar issue with post icon inside posts (one below post subject on left of post time). It is very small.

Everything else looks good.

VSEphpbb commented:
Didn't dig into find the issue, so here's a screenshot of a button I see that is messed up now by these changes: 
![screen shot 2015-08-30 at 2 06 43 pm](https://cloud.githubusercontent.com/assets/303711/9569436/6a6006c4-4f20-11e5-900c-55d2bb991eb4.png)

hanakin commented:
@VSEphpbb 
- I prefer plus unless anyone else has issues. 
- The close button is there for a reason I forget how but there was a weird bg issue with the icon when floating off the container. Ill look at it again. 
- I will look into the button issue, what browser is screenshot from?

@cyberalien 
- The icon color is slightly different because that was the color of the original icons. I can make them the same as the link if prefered. 
- icon sizes were chosen to match original icon sizes can adjust based on input I actually created three modifier classes for this reason. 
- Can you be more specific about orange icon screen shot as everything used is original colors.

VSEphpbb commented:
@hanakin That button issue is not browser related. It's just using the button2 class which you have removed, so it lost its formatting.

hanakin commented:
its fixed now

On Mon, Aug 31, 2015 at 8:07 AM, Matt Friedman <notifications@github.com>
wrote:

> @hanakin <https://github.com/hanakin> That button issue is not browser
> related. It's just using the button2 class which you have removed, so it
> lost its formatting.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phpbb/phpbb/pull/3851#issuecomment-136276817>.
>

cyberalien commented:
@hanakin 

Orange icon:
![icon](https://cloud.githubusercontent.com/assets/822287/9576169/1adbf824-4fde-11e5-9707-1c758461f7e4.png)

Small icon in post:
![icon2](https://cloud.githubusercontent.com/assets/822287/9576178/35c0a4f0-4fde-11e5-9e47-282fea64cf35.png)

I've tested all possible combinations of colors and FA glyphs for those icons, don't like any of them. Maybe keep these small icons as images instead of glyphs? Update images to have HD versions. Images look much better. FontAwesome file icons are too ugly at small font size.

hanakin commented:
i do not really see the problem the icon is not that different than the
originals.

On Mon, Aug 31, 2015 at 11:47 AM, Vjacheslav Trushkin <
notifications@github.com> wrote:

> @hanakin <https://github.com/hanakin>
>
> Orange icon:
> [image: icon]
> <https://cloud.githubusercontent.com/assets/822287/9576169/1adbf824-4fde-11e5-9707-1c758461f7e4.png>
>
> Small icon in post:
> [image: icon2]
> <https://cloud.githubusercontent.com/assets/822287/9576178/35c0a4f0-4fde-11e5-9e47-282fea64cf35.png>
>
> I've tested all possible combinations of colors and FA glyphs for those
> icons, don't like any of them. Maybe keep these small icons as images
> instead of glyphs? Update images to have HD versions. Images look much
> better. FontAwesome file icons are too ugly at small font size.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phpbb/phpbb/pull/3851#issuecomment-136317755>.
>

cyberalien commented:
But it is very different. This is current image:

![old](https://cloud.githubusercontent.com/assets/822287/9582438/080c8008-500e-11e5-81da-475df56f5055.png)

It is sharp, its color matches the rest of layout and has nice width/height ratio.

FA glyph is bland, blurred and looks fat. The more I look at it, the more I dislike it. It is very ugly. It might look nicer if its big, but when font size is tiny it looks like crap.

hanakin commented:
ok small fix to the notification drop-down. Swapped out the * for +. added slight alteration to the target post icon.
@cyberalien how about the icon now?
@VSEphpbb thoughts on the close button based on the attached screenshot of the issue?

anything else anyone sees?

VSEphpbb commented:
@hanakin :+1: close button looks good on the outer corner like that

hanakin commented:
k will update later tonight with those changes

On Thu, Sep 3, 2015 at 8:32 AM, Matt Friedman <notifications@github.com>
wrote:

> @hanakin <https://github.com/hanakin> [image: :+1:] close button looks
> good on the outer corner like that
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/phpbb/phpbb/pull/3851#issuecomment-137349049>.
>

hanakin commented:
@cyberalien when you get a chance can you look at this once more I think the target icon is the only thing left. unless anyone else has any objections. This should be ready to go.

cyberalien commented:
I don't see any difference.

cyberalien commented:
Never mind, its a bit bigger. But its still very blurred and orange.

hanakin commented:
It also narrower. I tweaked the scale slightly and fixed the color. should look pretty good now

hanakin commented:
@VSEphpbb @cyberalien can you check the target icon once more. 

I also replaced the contact menu icon and made some more changes to remove alot of the bloated icon/img based stuff from the theme

cyberalien commented:
Bug: icons in contact dropdown in messages are missing

Target icon: might be a bit better, but still very ugly. Now that I see you have SVG PR, maybe use something SVG instead of font?

Also its still orange for new posts. There is no orange color anywhere in style, so it doesn't belong there. It looked fine when it was just few pixels inside old icon, but if whole icon is orange it doesn't fit layout.

nickvergessen commented on phpBB/styles/prosilver/theme/stylesheet.css:
```diff
@@ -7,6 +7,7 @@
     --------------------------------------------------------------
 */
 
+@import url("https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css");
```
Should be added to our assets instead, so it also works if your forum is not connected to the internet or e.g. CDN is disabled?

hanakin replied:
agree, it will not even cache this way. We need a way to utilize the cdn when we can but also as you stated include it when no connection is present. I just did not know how the assets worked

nickvergessen commented on tests/functional/paging_test.php:
```diff
@@ -35,7 +35,7 @@ public function test_pagination()
 		$this->assertContains('post no16', $crawler->text());
 
-		$prev_link = $crawler->filter('.pagination > ul > li.previous > a')->attr('href');
+		$prev_link = $crawler->filter('.pagination > ul > li.prev > a')->attr('href');
```
unnecessary, or why can't we keep the old name?

hanakin replied:
sorry just common practice naming convention to use nxt and prev. can revert if need be

Nicofuma replied:
yes please, it will make the life of style author easier

nickvergessen commented on phpBB/styles/prosilver/template/viewtopic_body.html:
```diff
@@ -342,43 +373,47 @@ <h3 <!-- IF postrow.S_FIRST_ROW -->class="first"<!-- ENDIF -->><!-- IF postrow.P
-					<!-- DEFINE $QUICKMOD_AJAX = (quickmod.VALUE in ['lock', 'unlock', 'delete_topic', 'restore_topic', 'make_normal', 'make_sticky', 'make_announce', 'make_global']) -->
-					<li><a href="{quickmod.LINK}"<!-- IF $QUICKMOD_AJAX --> data-ajax="true" data-refresh="true"<!-- ENDIF -->>{quickmod.TITLE}</a></li>
-				<!-- END quickmod -->
+					<!-- BEGIN quickmod -->
```
please avoid retabbing to reduce the diff size of the style changes on release

hanakin replied:
This was due to tabbing be introduced early on that I had to fix and did not know what it look like before so I just fixed it all.

hanakin commented:
sorry do not see the problem with the icon choice looks fine too me, as for the color orange is used in pms for marked messages. If not orange what? unread color?

marc1706 commented:
The text inside post buttons do not seem to be centered. This looks especially weird since the icons in the post buttons are properly centered.

hanakin commented:
@marc1706 what buttons are you referring too? and in what browser? the post reply buttons or the post action buttons?

marc1706 commented:
I'm talking about these. The text is not properly vertically centered while the icons are.
![centered](https://cloud.githubusercontent.com/assets/394418/9756544/666ebc28-56db-11e5-95af-1f6be66ae739.PNG)
Browser is Chrome 45 on Windows.

hanakin commented:
the text is a pixel smaller and has always been i can increase it to the same as the icon, but the text has also always been aligned to the base of the icon not centered. 
![image](https://cloud.githubusercontent.com/assets/73081/9757321/1d40a78c-56e0-11e5-9a84-b80fd2915404.png)

marc1706 commented:
Hm, probably didn't notice it because the older icon was a little bit blurred

hanakin commented:
Ok update so everyone is aware of where this is at.

@cyberalien  I reverted all the previous changes to the small post icons and simplified them as this is a base theme we should stick to font-awsome for all icons if it does not work then thats probably a justification for removal of that icon or reworking the UI for said feature in the future.

@VSEphpbb I believe I have addressed all the previous issues

@nickvergessen just so we are on the same page as far the assets inclusion of font-awesome you are just wanting this included as a template variable similar to how jquery is included? This is the only thing I have left to work on for this pending any one else's issues I should be able to get to this by the end of the week.

hanakin commented:
@nickvergessen @Nicofuma I have fixed the inclusion of font-awesome for connection fallback but I do not think we should require this failing functional test. It makes absolutely no sense to test that all links are pointing at the styles directory to me. This just leads to hack js workarounds. Such as the one we are using to include the google font.

@cyberalien @VSEphpbb Can you please give this another look when you get time.

VSEphpbb commented:
@hanakin it's trying to load fonts locally from assets/fonts, which doesn't exist when CDN is off.

Also, for New Post/PM/Reply buttons, I think the `fa-pencil` icon will look a hell of a lot better, instead of plus sign? 
![screen shot 2015-09-13 at 9 29 09 pm](https://cloud.githubusercontent.com/assets/303711/9842131/85156628-5a5e-11e5-8d85-a615c9f61750.png)

Should the PM icon in the header be a `fa-envelope` instead of an inbox?
![screen shot 2015-09-13 at 9 30 25 pm](https://cloud.githubusercontent.com/assets/303711/9842139/ac73be22-5a5e-11e5-8ba0-d41dfb51f3b0.png)

VSEphpbb commented:
This jump boxes triangle needs to be colored correctly:
![screen shot 2015-09-13 at 9 32 51 pm](https://cloud.githubusercontent.com/assets/303711/9842160/1597fd78-5a5f-11e5-80c1-093420c761ba.png)

hanakin commented:
@VSEphpbb 

- pm icon choice is to avoid confusion as the envelope is used elsewhere for the contact us icon. 
- the pencil i do not think so as again to avoid confusion we are using the pencil for edit post.
- jump-box triangle is a seperate pr as thats not apart of this the issue with that is it needs its own class applied when used in that location as its an mcp jump which is rendered down instead of up like the normal jump.
- as for the CDN issue i see whats wrong and will fix that later tonight

VSEphpbb commented:
@hanakin I feel strongly about pencil though. It's not confusing as editing a post is the same result as writing a new post/pm... It leads you to the same thing, which is composing text.

VSEphpbb commented:
..adding to above, the New Topic (PM) buttons are the largest buttons we show. We should put our best foot forward with them by making them look nice. The plus sign is just crap. :-1:

VSEphpbb commented:
..and more. The edit post icon and the New Topic/PM buttons are never on the same page, so I see no confusion.

hanakin commented:
thats true they are not shown at the same time

VSEphpbb commented:
I don't think it'd be confusing to also use mail icon for PM notifications as we have currently been using the mail icon for PM notifications and the Contact link in 3.1, with no problems.

rxu commented:
Also, what if we'll have different icon colors for PMs/Notifications 0 and >0? F.e. for 0 it can be regular color and if you have PMs/Notifications it can be different (yellow/white/whatever suitable for the style).

VSEphpbb commented:
@hanakinfor the failing test, try adding `->eq(1)`

So in forum_style_test.php, replace all:
`$crawler->filter('head > link[rel=stylesheet]')->attr('href'));`
with:
`$crawler->filter('head > link[rel=stylesheet]')->eq(1)->attr('href'));`

This should skip over the first stylesheet link, which is now for fontawesome, and test the stylesheet for prosilver as before.

By the way, it would be good to add a similar test that does the same but checks to make sure fontawesome css is there.

```php
$this->assertContains('assets/css/font-awesome', $crawler->filter('head > link[rel=stylesheet]')->eq(0)->attr('href'));
```

hanakin commented:
@rxu outside the scope of this pr, but i would rather just not show it unless we have them. I never really seen the point of the 0 in the first place but thats maybe just me.

VSEphpbb commented:
Topic icons next to Private messages are missing in pm viewfolder.

hanakin commented:
Ok so I completely rebased for a cleaner diff to help avoid merge conflicts. So its going to need another round of reviews. Pending any other issues with the icon choices everything is good to go afterwards

cyberalien commented:
I still don't like mini post glyph, but if all images are supposed to be replaced, I don't see a better alternative.

Contact dropdown has gif icons instead of FA glyphs.

hanakin commented:
@cyberalien contact icons themselves are problematic for numerous reasons so I left it alone it will need to be its own pr as it will require a lot of rewriting in the backend as everything is include based on individual DB values it may be better handleded as svgs rather than font as well

cyberalien commented:
Sounds good.

Then everything seems fine. I don't see any errors.

marc1706 commented:
I get three 404s:
```
GET http://localhost/phpbb3/assets/fonts/fontawesome-webfont.woff2?v=4.4.0 
GET http://localhost/phpbb3/assets/fonts/fontawesome-webfont.woff?v=4.4.0 
GET http://localhost/phpbb3/assets/fonts/fontawesome-webfont.ttf?v=4.4.0
```

Nicofuma commented:
I don't have these 404. @marc1706 can you check please?

Otherwise looks good

Nicofuma commented:
@cyberalien By the way, do you know/have an idea how much work it will require to have prosilver se compatible?

hanakin commented:
@Nicofuma the 404 was fixed by https://github.com/hanakin/phpbb/commit/ff6d590b3693c5a20f1626664bb92bb18f384d3a
